crates/cargo-contract/src/cmd/info.rs

```diff
@@ -55,75 +62,90 @@ pub struct InfoCommand {
+    /// Display all contracts addresses
+    #[clap(name = "all", long)]
+    all: bool,
 }
 
 impl InfoCommand {
     pub fn run(&self) -> Result<(), ErrorVariant> {
-        tracing::debug!(
-            "Getting contract information for AccountId {:?}",
-            self.contract
-        );
-
         Runtime::new()?.block_on(async {
             let url = self.url.clone();
             let client = OnlineClient::<DefaultConfig>::from_url(url).await?;
 
-            let info_result = fetch_contract_info(&self.contract, &client).await?;
-
-            match info_result {
-                Some(info_to_json) => {
-                    match (self.output_json, self.binary) {
-                        (true, false) => println!("{}", info_to_json.to_json()?),
-                        (false, false) => {
-                            basic_display_format_contract_info(&info_to_json)
-                        }
-                        // Binary flag applied
-                        (_, true) => {
-                            let wasm_code =
-                                fetch_wasm_code(*info_to_json.code_hash(), &client)
-                                    .await?;
-                            match (wasm_code, self.output_json) {
-                                (Some(code), false) => {
-                                    std::io::stdout()
-                                        .write_all(&code)
-                                        .expect("Writing to stdout failed")
-                                }
-                                (Some(code), true) => {
-                                    let wasm = serde_json::json!({
-                                        "wasm": format!("0x{}", hex::encode(code))
-                                    });
-                                    println!(
-                                        "{}",
-                                        serde_json::to_string_pretty(&wasm).map_err(
-                                            |err| {
-                                                anyhow!(
-                                                    "JSON serialization failed: {}",
-                                                    err
-                                                )
-                                            }
-                                        )?
-                                    );
-                                }
-                                (None, _) => {
-                                    return Err(anyhow!(
-                                        "Contract wasm code was not found"
-                                    )
-                                    .into())
-                                }
-                            }
-                        }
+            // All flag applied
+            if self.all {
+                // Using count of 255 to avoid conversion issue to usize on
+                // any architecture
+                let count = 255;
```

Comment:
Could be a `const COUNT: u32 = 255;`

Reply:
Or even `const COUNT: u8 = 255` if we are not assuming 32 bit platform (although we may already assume min 32 bits in other places). Then use the infallible `into()` for converting to `u32`.

Reply:
I see that we are using even u64 to usize conversion so I will keep count as u32 and use `count as usize`. Thank you for the hint.

Reply:
Done